src/uct/ugni/smsg/ugni_smsg_iface.c

```diff
@@ -265,19 +265,33 @@ static ucs_status_t uct_ugni_smsg_ep_flush(uct_ep_h tl_ep, unsigned flags,
                                            uct_completion_t *comp)
 {
     uct_ugni_smsg_ep_t *ep = ucs_derived_of(tl_ep, uct_ugni_smsg_ep_t);
+    ucs_status_t status;
 
-    if (comp != NULL) {
-        return UCS_ERR_UNSUPPORTED;
+    /* Even if we have no outstanding requests we can still get back send credits */
+    status = progress_local_cq(ucs_derived_of(tl_ep->iface, uct_ugni_smsg_iface_t));
+    
+    if ((0 == ep->super.outstanding) && uct_ugni_can_send(&ep->super)) {
+        UCT_TL_IFACE_STAT_FLUSH(ucs_derived_of(tl_ep->iface, uct_base_iface_t));
+        status = UCS_OK;
+    } else if (comp != NULL) {
+        status = UCS_ERR_NO_RESOURCE;
```

Comment:
Currently, it will return INPROGRESS. It seems better to say INPROGRESS when there is no comp passed in and there are still outstanding requests on the EP.